rustls/src/msgs/deframer.rs

```diff
@@ -84,8 +84,11 @@ impl MessageDeframer {
                             InvalidMessage::UnknownProtocolVersion
                         }
                     };
-                    self.desynced = true;
-                    return Err(Error::CorruptMessage(err_kind));
+
+                    return Err(self
+                        .last_error
+                        .insert(Error::CorruptMessage(err_kind))
+                        .clone());
```

Comment:
Hmm, I would maybe go with `self.last_error = Some(Error::from(err_kind)); return Err(Error::from(err_kind));`, preferring the slight duplication over the verbosity of this construction (which is might've been nice if it didn't get spread over 4 lines).

Reply:
Does this feedback apply to all callsites that are currently using the `.insert()` pattern? I had originally done it with an intermediate everywhere but it got kind of cluttered.

Reply:
Hmm, why was it cluttered? Maybe one we derive `Copy` we can away without the `.clone()` call (might need a deref operator) so things become a little cleaner?

Reply:
I ended up trying out a very small helper function, `set_err`, that encloses this convert, assign, return pattern the deframer uses in multiple places. The shortness of the function allowed all of the callsites to be formatted pretty cleanly, IMO. Please take a look and see if that also works for you.